developers_chamber/scripts/qa.py

```diff
@@ -0,0 +1,246 @@
+    """
+    name = None
+
+    def _get_repo(self):
+        """
+        Returns the repo object.
+        """
+        return Repo('.')
+
+    def _get_default_branch(self):
+        """
+        Returns default branch of the repo.
+        """
+        return self._run_command("git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@'")
+
+    def _get_diffs(self):
+        """
+        Returns Git diffs against default branch.
+        """
+        repo = self._get_repo()
+        target_commit = repo.heads[self._get_default_branch()].commit
+        return target_commit.diff('HEAD', create_patch=True)
+
+    def _get_unstaged(self):
+        """
+        Returns unstaged files in the repo.
+        """
+        return self._get_repo().index.diff(None, create_patch=True)
+
+    def _run_command(self, command):
+        """
+        Runs shell command and returns its output.
+        """
+        try:
+            return subprocess.check_output(command, shell=True).decode().strip()
+        except subprocess.CalledProcessError as ex:
+            raise QAError(str(ex), ex.output.decode())
+
+    def _run_check(self):
+        """
+        This function should implement the actual check and should raise `QAError` if failed.
+        """
+        raise NotImplementedError()
+
+    def _cleanup(self):
+        """
+        Cleans up the repo to be fresh for another check.
+        """
+        self._get_repo().git.stash('save')
+
+    def run(self):
+        """
+        Runs the check and cleanup methods.
+        """
+        self._run_check()
+        self._cleanup()
+
+
+class QACheckRunner:
+    """
+    Runs multiple checks and evaluates the results while printing a nice output.
+    """
+    def __init__(self, *checks):
+        """
+        Arguments:
+            checks: Instances of QACheck class.
+        """
+        self.checks = checks
+
+    def run(self):
+        print('-----------------------------------------------------------')
+        print_bold(' Quality Assurance Check')
+        print('-----------------------------------------------------------')
+
+        failed_checks = []
+        for i, check in enumerate(self.checks):
+            print_heading('[{}/{}] {}...'.format(i + 1, len(self.checks), check.name))
+            try:
+                check.run()
+            except QAError as ex:
+                failed_checks.append(check)
+                print_error(ex)
+                print(ex.output)
+            else:
+                print_success('OK')
+
+        print('-----------------------------------------------------------')
+
+        if failed_checks:
+            print_error('FAILURE: {} check(s) failed!'.format(len(failed_checks)))
+            exit_code = 1
+        else:
+            print_success('SUCCESS: All QA checks passed!')
+            exit_code = 0
+
+        print('-----------------------------------------------------------')
+        exit(exit_code)
```

Comment:
Why don't you handle it through Click's exceptions?

Reply:
I've never worked with click, thanks for the suggestion, I'll check it out.

Reply:
I think in this case, it's better to explicitly set exit status since i want to have control over it.